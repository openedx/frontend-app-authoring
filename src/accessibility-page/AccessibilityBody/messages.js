import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  a11yBodyPolicyLink: {
    id: 'a11yBodyPolicyLink',
    defaultMessage: 'Website Accessibility Policy',
    description: 'Title for link to full accessibility policy.',
  },
  a11yBodyPageHeader: {
    id: 'a11yBodyPageHeader',
    defaultMessage: 'Individualized Accessibility Process for Course Creators',
    description: 'Heading for studio\'s accessibility policy page.',
  },
  a11yBodyIntroGraph: {
    id: 'a11yBodyIntroGraph',
    defaultMessage: `At edX, we seek to understand and respect the unique needs and perspectives of the edX global community.
      We value every course team and are committed to expanding access to all, including course team creators and authors with
      disabilities. To that end, we have adopted a {communityAccessibilityLink} and this process to allow course team creators
      and authors to request assistance if they are unable to develop and post content on our platform via Studio because of their
      disabilities.`,
    description: 'Introductory paragraph outlining why we care about accessibility, and what we\'re doing about it.',
  },
  a11yBodyStepsHeader: {
    id: 'a11yBodyStepsHeader',
    defaultMessage: 'Course team creators and authors needing such assistance should take the following steps:',
    description: 'Heading for list of steps authors can take for accessibility requests.',
  },
  a11yBodyEdxResponse: {
    id: 'a11yBodyEdxResponse',
    defaultMessage: `'We will communicate with you about your preferences and needs in determining the appropriate solution, although
      the ultimate decision will be ours, provided that the solution is effective and timely. The factors we will consider in choosing
      an accessibility solution are: effectiveness; timeliness (relative to your deadlines); ease of implementation; and ease of use for
      you. We will notify you of the decision and explain the basis for our decision within 10 business days of discussing with you.`,
    description: 'Paragraph outlining how we will select an accessibility solution.',
  },
  a11yBodyEdxFollowUp: {
    id: 'a11yBodyEdxFollowUp',
    defaultMessage: `Thereafter, we will communicate with you on a weekly basis regarding our evaluation, decision, and progress in
      implementing the accessibility solution. We will notify you when implementation of your accessibility solution is complete and
      will follow-up with you as may be necessary to see if the solution was effective.`,
    description: 'Paragraph outlining how we will follow-up with you during and after implementing an accessibility solution.',
  },
  a11yBodyOngoingSupport: {
    id: 'a11yBodyOngoingSupport',
    defaultMessage: 'EdX will provide ongoing technical support as needed and will address any additional issues that arise after the initial course creation.',
    description: 'A statement of ongoing support.',
  },
  a11yBodyA11yFeedback: {
    id: 'a11yBodyA11yFeedback',
    defaultMessage: 'Please direct any questions or suggestions on how to improve the accessibility of Studio to {emailElement} or use the form below. We welcome your feedback.',
    description: 'Contact information heading for those with accessibility issues or suggestions.',
  },
  a11yBodyEmailHeading: {
    id: 'a11yBodyEmailHeading',
    defaultMessage: 'Send an email to {emailElement} with the following information:',
    description: 'Heading for list of information required when you email us.',
  },
  a11yBodyNameEmail: {
    id: 'a11yBodyNameEmail',
    defaultMessage: 'your name and email address;',
    description: 'Your contact information.',
  },
  a11yBodyInstitution: {
    id: 'a11yBodyInstitution',
    defaultMessage: 'the edX member institution that you are affiliated with;',
    description: 'edX affiliate information.',
  },
  a11yBodyBarrier: {
    id: 'a11yBodyBarrier',
    defaultMessage: 'a brief description of the challenge or barrier to access that you are experiencing; and',
    description: 'Accessibility problem information.',
  },
  a11yBodyTimeConstraints: {
    id: 'a11yBodyTimeConstraints',
    defaultMessage: 'how soon you need access and for how long (e.g., a planned course start date or in connection with a course-related deadline such as a final essay).',
    description: 'Time contstraint information.',
  },
  a11yBodyReceipt: {
    id: 'a11yBodyReceipt',
    defaultMessage: 'The edX Support Team will respond to confirm receipt and forward your request to the edX Partner Manager for your institution and the edX Website Accessibility Specialist.',
    description: 'Paragraph outlining what steps edX will take immediately.',
  },
  a11yBodyExtraInfo: {
    id: 'a11yBodyExtraInfo',
    defaultMessage: `With guidance from the Website Accessibility Specialist, edX will contact you to discuss your request and gather
      additional information from you about your preferences and needs, to determine if there's a workable solution that edX is able to support.`,
    description: 'Paragraph outlining how and when edX will reach out to you.',
  },
  a11yBodyFixesListHeader: {
    id: 'a11yBodyFixesListHeader',
    defaultMessage: 'EdX will assist you promptly and thoroughly so that you are able to create content on the CMS within your time constraints.  Such efforts may include, but are not limited to:',
    description: 'Heading for list of ways we might be able to assist.',
  },
  a11yBodyThirdParty: {
    id: 'a11yBodyThirdParty',
    defaultMessage: 'Purchasing a third-party tool or software for use on an individual basis to assist your use of Studio;',
    description: 'Buy third-party software.',
  },
  a11yBodyContractor: {
    id: 'a11yBodyContractor',
    defaultMessage: 'Engaging a trained independent contractor to provide real-time visual, verbal and physical assistance; or',
    description: 'Hire a contractor.',
  },
  a11yBodyCodeFix: {
    id: 'a11yBodyCodeFix',
    defaultMessage: 'Developing new code to implement a technical fix.',
    description: 'Make a technical fix.',
  },
});

export default messages;
