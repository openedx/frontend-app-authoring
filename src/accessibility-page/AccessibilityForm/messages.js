import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  accessibilityPolicyFormEmailLabel: {
    id: 'accessibilityPolicyFormEmailLabel',
    defaultMessage: 'Email Address',
    description: 'Label for the email form field',
  },
  accessibilityPolicyFormErrorHighVolume: {
    id: 'accessibilityPolicyFormErrorHighVolume',
    defaultMessage: 'We are currently experiencing high volume. Try again later today or send an email message to {emailLink}.',
    description: 'Error message when site is experiencing high volume that will include an email link',
  },
  accessibilityPolicyFormErrorMissingFields: {
    id: 'accessibilityPolicyFormErrorMissingFields',
    defaultMessage: 'Make sure to fill in all fields.',
    description: 'Error message to instruct user to fill in all fields',
  },
  accessibilityPolicyFormHeader: {
    id: 'accessibilityPolicyFormHeader',
    defaultMessage: 'Studio Accessibility Feedback',
    description: 'The heading for the form',
  },
  accessibilityPolicyFormMessageLabel: {
    id: 'accessibilityPolicyFormMessageLabel',
    defaultMessage: 'Message',
    description: 'Label for the message form field',
  },
  accessibilityPolicyFormNameLabel: {
    id: 'accessibilityPolicyFormNameLabel',
    defaultMessage: 'Name',
    description: 'Label for the name form field',
  },
  accessibilityPolicyFormSubmitAria: {
    id: 'accessibilityPolicyFormSubmitAria',
    defaultMessage: 'Submit Accessibility Feedback Form',
    description: 'Detailed aria-label for the submit button',
  },
  accessibilityPolicyFormSubmitLabel: {
    id: 'accessibilityPolicyFormSubmitLabel',
    defaultMessage: 'Submit',
    description: 'General label for the submit button',
  },
  accessibilityPolicyFormSubmittingFeedbackLabel: {
    id: 'accessibilityPolicyFormSubmittingFeedbackLabel',
    defaultMessage: 'Submitting',
    description: 'Loading message while form feedback is being submitted',
  },
  accessibilityPolicyFormSuccess: {
    id: 'accessibilityPolicyFormSuccess',
    defaultMessage: 'Thank you for contacting edX!',
    description: 'Simple thank you message when form submission is successful',
  },
  accessibilityPolicyFormSuccessDetails: {
    id: 'accessibilityPolicyFormSuccessDetails',
    defaultMessage: 'Thank you for your feedback regarding the accessibility of Studio. We typically respond within one business day ({day_start} to {day_end}, {time_start} to {time_end}).',
    description: 'Detailed thank you message when form submission is successful',
  },
  accessibilityPolicyFormValidEmail: {
    id: 'accessibilityPolicyFormValidEmail',
    defaultMessage: 'Enter a valid email address.',
    description: 'Error message for when an invalid email is entered into the form',
  },
  accessibilityPolicyFormValidMessage: {
    id: 'accessibilityPolicyFormValidMessage',
    defaultMessage: 'Enter a message.',
    description: 'Error message an invalid message is entered into the form',
  },
  accessibilityPolicyFormValidName: {
    id: 'accessibilityPolicyFormValidName',
    defaultMessage: 'Enter a name.',
    description: 'Error message an invalid name is entered into the form',
  },
});

export default messages;
