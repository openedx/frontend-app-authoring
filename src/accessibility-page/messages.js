import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  pageTitle: {
    id: 'course-authoring.import.page.title',
    defaultMessage: 'Studio Accessibility Policy| {siteName}',
  },
});

export default messages;
