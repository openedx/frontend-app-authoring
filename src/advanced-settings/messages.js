import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  headingTitle: {
    id: 'course-authoring.advanced-settings.heading.title',
    defaultMessage: 'Advanced settings',
  },
  headingSubtitle: {
    id: 'course-authoring.advanced-settings.heading.subtitle',
    defaultMessage: 'Settings',
  },
  policy: {
    id: 'course-authoring.advanced-settings.policies.title',
    defaultMessage: 'Manual policy definition',
  },
  alertWarning: {
    id: 'course-authoring.advanced-settings.alert.warning',
    defaultMessage: "You've made some changes",
  },
  alertWarningDescriptions: {
    id: 'course-authoring.advanced-settings.alert.warning.descriptions',
    defaultMessage: 'Your changes will not take effect until you save your progress. Take care with key and value formatting, as validation is not implemented.',
  },
  alertSuccess: {
    id: 'course-authoring.advanced-settings.alert.success',
    defaultMessage: 'Your policy changes have been saved.',
  },
  alertSuccessDescriptions: {
    id: 'course-authoring.advanced-settings.alert.success.descriptions',
    defaultMessage: 'No validation is performed on policy keys or value pairs. If you are having difficulties, check your formatting.',
  },
  alertProctoringError: {
    id: 'course-authoring.advanced-settings.alert.proctoring.error',
    defaultMessage: 'This course has protected exam setting that are incomplete or invalid.',
  },
  alertProctoringErrorDescriptions: {
    id: 'course-authoring.advanced-settings.alert.proctoring.error.descriptions',
    defaultMessage: 'You will be unable to make changes until the following setting are updated on the page below.',
  },
  buttonSaveText: {
    id: 'course-authoring.advanced-settings.alert.button.save',
    defaultMessage: 'Save changes',
  },
  buttonSavingText: {
    id: 'course-authoring.advanced-settings.alert.button.saving',
    defaultMessage: 'Saving',
  },
  buttonCancelText: {
    id: 'course-authoring.advanced-settings.alert.button.cancel',
    defaultMessage: 'Cancel',
  },
  deprecatedButtonShowText: {
    id: 'course-authoring.advanced-settings.deprecated.button.show',
    defaultMessage: 'Show',
  },
  deprecatedButtonHideText: {
    id: 'course-authoring.advanced-settings.deprecated.button.hide',
    defaultMessage: 'Hide',
  },
  alertWarningAriaLabelledby: {
    id: 'course-authoring.advanced-settings.alert.warning.aria.labelledby',
    defaultMessage: 'notification-warning-title',
  },
  alertWarningAriaDescribedby: {
    id: 'course-authoring.advanced-settings.alert.warning.aria.describedby',
    defaultMessage: 'notification-warning-description',
  },
  alertSuccessAriaLabelledby: {
    id: 'course-authoring.advanced-settings.alert.success.aria.labelledby',
    defaultMessage: 'alert-confirmation-title',
  },
  alertSuccessAriaDescribedby: {
    id: 'course-authoring.advanced-settings.alert.success.aria.describedby',
    defaultMessage: 'alert-confirmation-description',
  },
  alertProctoringAriaLabelledby: {
    id: 'course-authoring.advanced-settings.alert.proctoring.error.aria.labelledby',
    defaultMessage: 'alert-danger-title',
  },
  alertProctoringDescribedby: {
    id: 'course-authoring.advanced-settings.alert.proctoring.error.aria.describedby',
    defaultMessage: 'alert-danger-description',
  },
});

export default messages;
