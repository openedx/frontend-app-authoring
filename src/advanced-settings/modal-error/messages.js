import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  modalErrorTitle: {
    id: 'course-authoring.advanced-settings.modal.error.title',
    defaultMessage: 'Validation error while saving',
  },
  modalErrorButtonChangeManually: {
    id: 'course-authoring.advanced-settings.modal.error.btn.change-manually',
    defaultMessage: 'Change manually',
  },
  modalErrorButtonUndoChanges: {
    id: 'course-authoring.advanced-settings.modal.error.btn.undo-changes',
    defaultMessage: 'Undo changes',
  },
});

export default messages;
