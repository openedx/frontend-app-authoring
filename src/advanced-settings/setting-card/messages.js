import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  deprecated: {
    id: 'course-authoring.advanced-settings.button.deprecated',
    defaultMessage: 'Deprecated',
  },
  helpButtonText: {
    id: 'course-authoring.advanced-settings.button.help',
    defaultMessage: 'Show help text',
  },
});

export default messages;
