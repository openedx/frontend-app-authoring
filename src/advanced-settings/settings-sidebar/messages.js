import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  about: {
    id: 'course-authoring.advanced-settings.sidebar.about.title',
    defaultMessage: 'What do advanced settings do?',
  },
  aboutDescription1: {
    id: 'course-authoring.advanced-settings.sidebar.about.description-1',
    defaultMessage: 'Advanced settings control specific course functionality. On this page, you can edit manual policies, which are JSON-based key and value pairs that control specific course settings.',
  },
  aboutDescription2: {
    id: 'course-authoring.advanced-settings.sidebar.about.description-2',
    defaultMessage: 'Any policies you modify here override all other information youâ€™ve defined elsewhere in Studio. Do not edit policies unless you are familiar with both their purpose and syntax.',
  },
  other: {
    id: 'course-authoring.advanced-settings.sidebar.other.title',
    defaultMessage: 'Other course settings',
  },
  otherCourseSettingsLinkToScheduleAndDetails: {
    id: 'course-authoring.advanced-settings.sidebar.links.schedule-and-details',
    defaultMessage: 'Details & schedule',
    description: 'Link to Studio Details & schedule page',
  },
  otherCourseSettingsLinkToGrading: {
    id: 'course-authoring.advanced-settings.sidebar.links.grading',
    defaultMessage: 'Grading',
    description: 'Link to Studio Grading page',
  },
  otherCourseSettingsLinkToCourseTeam: {
    id: 'course-authoring.advanced-settings.sidebar.links.course-team',
    defaultMessage: 'Course team',
    description: 'Link to Studio Course team page',
  },
  otherCourseSettingsLinkToGroupConfigurations: {
    id: 'course-authoring.advanced-settings.sidebar.links.group-configurations',
    defaultMessage: 'Group configurations',
    description: 'Link to Studio Group configurations page',
  },
  otherCourseSettingsLinkToProctoredExamSettings: {
    id: 'course-authoring.advanced-settings.sidebar.links.proctored-exam-settings',
    defaultMessage: 'Proctored exam settings',
    description: 'Link to Proctored exam settings page',
  },
});

export default messages;
