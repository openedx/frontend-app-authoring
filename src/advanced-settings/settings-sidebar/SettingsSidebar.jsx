import React from 'react';
import {
  FormattedMessage,
  injectIntl,
  intlShape,
} from '@edx/frontend-platform/i18n';
import PropTypes from 'prop-types';

import { HelpSidebar } from '../../generic/help-sidebar';
import messages from './messages';

const SettingsSidebar = ({ intl, courseId, proctoredExamSettingsUrl }) => (
  <HelpSidebar
    courseId={courseId}
    proctoredExamSettingsUrl={proctoredExamSettingsUrl}
    showOtherSettings
  >
    <h4 className="help-sidebar-about-title">
      {intl.formatMessage(messages.about)}
    </h4>
    <p className="help-sidebar-about-descriptions">
      {intl.formatMessage(messages.aboutDescription1)}
    </p>
    <p className="help-sidebar-about-descriptions">
      {intl.formatMessage(messages.aboutDescription2)}
    </p>
    <p className="help-sidebar-about-descriptions">
      <FormattedMessage
        id="course-authoring.advanced-settings.about.description-3"
        defaultMessage="{notice} When you enter strings as policy values, ensure that you use double quotation marks (“) around the string. Do not use single quotation marks (‘)."
        values={{ notice: <strong>Note:</strong> }}
      />
    </p>
  </HelpSidebar>
);

SettingsSidebar.defaultProps = {
  proctoredExamSettingsUrl: '',
};

SettingsSidebar.propTypes = {
  intl: intlShape.isRequired,
  courseId: PropTypes.string.isRequired,
  proctoredExamSettingsUrl: PropTypes.string,
};

export default injectIntl(SettingsSidebar);
