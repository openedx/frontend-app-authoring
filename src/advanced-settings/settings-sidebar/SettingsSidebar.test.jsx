import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import { initializeMockApp } from '@edx/frontend-platform';
import { AppProvider } from '@edx/frontend-platform/react';

import initializeStore from '../../store';
import SettingsSidebar from './SettingsSidebar';
import messages from './messages';

const courseId = 'course-123';
let store;

const RootWrapper = () => (
  <AppProvider store={store}>
    <IntlProvider locale="en" messages={{}}>
      <SettingsSidebar intl={{ formatMessage: jest.fn() }} courseId={courseId} />
    </IntlProvider>
  </AppProvider>
);

describe('<SettingsSidebar />', () => {
  beforeEach(() => {
    initializeMockApp({
      authenticatedUser: {
        userId: 3,
        username: 'abc123',
        administrator: true,
        roles: [],
      },
    });
    store = initializeStore();
  });
  it('renders about and other sidebar titles correctly', () => {
    const { getByText } = render(<RootWrapper />);
    expect(getByText(messages.about.defaultMessage)).toBeInTheDocument();
    expect(getByText(messages.other.defaultMessage)).toBeInTheDocument();
  });
  it('renders about descriptions correctly', () => {
    const { getByText } = render(<RootWrapper />);
    const aboutThirtyDescription = getByText('When you enter strings as policy values, ensure that you use double quotation marks (“) around the string. Do not use single quotation marks (‘).');
    expect(getByText(messages.aboutDescription1.defaultMessage)).toBeInTheDocument();
    expect(getByText(messages.aboutDescription2.defaultMessage)).toBeInTheDocument();
    expect(aboutThirtyDescription).toBeInTheDocument();
  });
});
