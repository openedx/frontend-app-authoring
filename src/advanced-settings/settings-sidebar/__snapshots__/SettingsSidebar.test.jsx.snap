// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsSidebar should match the snapshot 1`] = `
<aside
  className="help-sidebar"
>
  <div
    className="help-sidebar-about"
  >
    <h4
      className="help-sidebar-about-title"
    >
      What do advanced settings do?
    </h4>
    <p
      className="help-sidebar-about-descriptions"
    >
      Advanced settings control specific course functionality. On this page, you can edit manual policies, which are JSON-based key and value pairs that control specific course settings.
    </p>
    <p
      className="help-sidebar-about-descriptions"
    >
      Any policies you modify here override all other information you’ve defined elsewhere in Studio. Do not edit policies unless you are familiar with both their purpose and syntax.
    </p>
    <p
      className="help-sidebar-about-descriptions"
    >
      <strong>
        Note:
      </strong>
       When you enter strings as policy values, ensure that you use double quotation marks (“) around the string. Do not use single quotation marks (‘).
    </p>
  </div>
  <hr />
  <div
    className="help-sidebar-other"
  >
    <h4
      className="help-sidebar-other-title"
    >
      Other Course Settings
    </h4>
    <nav
      aria-label="Other Course Settings"
      className="help-sidebar-other-links"
    >
      <ul
        className="p-0 mb-0"
      >
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="http://localhost:2001/course/course123/settings/details"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Schedule & Details
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="http://localhost:2001/course/course123/settings/grading"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Grading
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="http://localhost:2001/course/course123/course_team"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Course Team
          </a>
        </li>
        
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="https://example.com/group_configurations/course123"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Group Configurations
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="http://localhost:2001/course/course123/settings/advanced"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Advanced Settings
          </a>
        </li>
      </ul>
    </nav>
  </div>
</aside>
`;
